src/modules/products/products.service.ts

```diff
@@ -0,0 +1,56 @@
+import { BadRequestException, Injectable, InternalServerErrorException} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Product } from './entities/product.entity';
+import { PaginationResult } from './interface/PaginationInterface';
+
+@Injectable()
+export class ProductsService {
+  constructor(
+    @InjectRepository(Product)
+    private productRepository: Repository<Product>
+  ) {}
+
+  async findAllProducts(page: number = 1, limit: number = 5): Promise<PaginationResult> {
+    try {
+      const [products, totalCount] = await this.productRepository.findAndCount({
+        take: limit,
+        skip: limit * (page - 1),
+      });
+      if (limit !== undefined && (isNaN(Number(limit)) || Number(limit) < 0)) {
+        throw new BadRequestException({
+          status: 'bad request',
+          message: 'Invalid query params passed',
+          status_code: 400,
+        });
+      }
+      if (page !== undefined && (isNaN(Number(page)) || Number(page) < 0)) {
+        throw new BadRequestException({
+          status: 'bad request',
+          message: 'Invalid query params passed',
+          status_code: 400,
+        });
+      }
```

Comment:
You should query the dB after the checks are done.

```suggestion
if (
  (limit !== undefined && (isNaN(limitNumber) || limitNumber < 0)) ||
  (page !== undefined && (isNaN(pageNumber) || pageNumber < 0))
) {
  throw new BadRequestException({
    status: 'bad request',
    message: 'Invalid query params passed',
    status_code: 400,
  });
}

const [products, totalCount] = await this.productRepository.findAndCount({
  take: limitNumber,
  skip: limitNumber * (pageNumber - 1),
});
```

Reply:
Done